### PR TITLE
Add converter for objects such as Monitor

### PR DESCRIPTION
**Pull request by t-wissmann**
This introduces a new converter class for the parsing of descriptions of
objects, e.g. Monitors, Tags, Clients. So far, only the converter for
Monitor* is implemented. If a command gets a monitor as a parameter, we
can now use ArgParse which internally uses the Converter class.

This new converter is used in the commands remove_monitor and tag_status
and their test cases have slightly been adjusted and extended.

### CODE REVIEW

File: tests/test_monitor.py [hunk 1]
```diff
@@ -85,9 +85,23 @@ def test_remove_monitor(hlwm):
...
+    hlwm.call('add_monitor 800x600+40+40 tag2 monitor2')
+
+    hlwm.call_xfail('remove_monitor foobar') \
+        .expect_stderr('No such monitor: foobar')
+
+    hlwm.call_xfail('remove_monitor 2') \
+        .expect_stderr('No such monitor: 2')
+
+
 def test_cannot_remove_nonexistent_monitor(hlwm):
```
Comment (blarz) [inline tests/test_monitor.py:102]: Isn't this test a bit redundant now?

**Review by blarz — COMMENTED** [review]
Just a tiny question regarding tests

File: tests/test_monitor.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_monitor.py]: Right,.. I've removed it and added asserts in the `test_remove_monitor_invalid_args`

**Comment by blarz** [comment]
How is the ArgParse distribution state btw? Is it used "everywhere"? I think a saw a issue for removing the old version of it?

**Comment by blarz** [comment]
> How is the ArgParse distribution state btw? Is it used "everywhere"? I think a saw a issue for removing the old version of it?

Ah, I confused that one, I meant  #136

File: tests/test_monitor.py (hunk 1 above)
Comment (blarz) [reply tests/test_monitor.py]: Thanks!

**Review by blarz — APPROVED** [review]

**Comment by t-wissmann** [comment]
> How is the ArgParse distribution state btw? Is it used "everywhere"? I think a saw a issue for removing the old version of it?

It's not used everywhere but I'd like to do so.

**Pull request merged**